akka-stream/src/main/scala/akka/stream/scaladsl/Queue.scala

```diff
@@ -32,6 +32,23 @@ trait SourceQueue[T] {
 }
 
 /**
+ * This trait adds completion support to [[SourceQueue]].
+ */
+trait SourceQueueWithComplete[T] extends SourceQueue[T] {
+  /**
+   * Complete the stream normally. Use `watchCompletion` to be notified of this
+   * operation’s success.
+   */
+  def complete(): Unit
+
+  /**
+   * Complete the stream with a failure. Use `watchCompletion` to be notified of this
+   * operation’s success.
+   */
+  def fail(ex: Throwable): Unit
```

Comment:
This leaves me even more confused. Do we agree now or don’t we? I don’t think there is anything to fix here because declaring this kind of problem worthy of a fix would be inconsistent with many other occasions where we will not or cannot do the equivalent thing.